File: 1137. N-th Tribonacci Number.java
class Solution {
    public int tribonacci(int n) {
        if (n == 0) return 0;
        if (n == 1 || n == 2) return 1;
        int a = 0, b = 1, c = 1;
        while (n-- >= 3) {
            int next = a + b + c;
            a = b;
            b = c;
            c = next;
        }
        return c;
    }
}

File: go/1137. N-th Tribonacci Number.go
func tribonacci(n int) int {
	if n == 0 {
		return 0
	}
	if n == 1 || n == 2 {
		return 1
	}
	a, b, c := 0, 1, 1
	for n >= 3 {
		c, b, a = a+b+c, c, b
		n -= 1
	}
	return c
}